### PR TITLE
feature: database (db) migration, seed, environments and other code c…

### PR DESCRIPTION
**Pull request by hhrynenko**
feature: database (db) migration, seed, environments and other code changes for new db logic.

### CODE REVIEW

File: .env-example [hunk 1]
```diff
@@ -1,9 +1,15 @@
 PORT=
 # DATABASE_PARAMS
-DB_NAME=
```
Comment (stovpak) [inline .env-example:3]: rollback

File: .sequelizerc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const path = require('path');
+
+module.exports = {
+  'config': path.resolve('config', 'config.js'),
+  'models-path': path.resolve('db', 'models'),
+  'seeders-path': path.resolve('db', 'seeders'),
+  'migrations-path': path.resolve('db', 'migrations')
+};
```
Comment (stovpak) [inline .sequelizerc]: add new line

File: config/config.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    production: {
+        username: process.env.PROD_DB_USER,
+        password: process.env.PROD_DB_PASSWORD,
+        database: process.env.PROD_DB_NAME,
+        host: process.env.DB_HOST,
+        port: process.env.DB_PORT,
+        dialect: 'postgres',
+    },
+};
```
Comment (stovpak) [inline config/config.js:12]: you shoud have 1 db env, maximum 2 (real and test)

File: db/migrations/20220806190612-create-city.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('city', {
```
Comment (stovpak) [inline db/migrations/20220806190612-create-city.js]: add checks
related to every mutation

File: db/models/city.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const {
+  Model,
+} = require('sequelize');
+
+module.exports = (sequelize, DataTypes) => {
+  class City extends Model {
+    /**
+     * Helper method for defining associations.
```
Comment (stovpak) [inline db/models/city.js]: remove comments

File: db/models/comment.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+const {
+  Model,
+} = require('sequelize');
+
+module.exports = (sequelize, DataTypes) => {
+  class Comment extends Model {
+    /**
+     * Helper method for defining associations.
```
Comment (stovpak) [inline db/models/comment.js]: the same

File: db/models/index.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+const path = require('path');
+const Sequelize = require('sequelize');
+
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(`${__dirname}/../../config/config.js`)[env];
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
```
Comment (stovpak) [inline db/models/index.js]: remove check

File: db/seeders/20220810083343-demo-cities.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    await queryInterface.bulkInsert('city', [
+      { id: 1, cityName: 'City1', averageGrade: 7 },
+      { id: 2, cityName: 'City2', averageGrade: 5 },
+      { id: 3, cityName: 'City3', averageGrade: 4 },
+      { id: 4, cityName: 'City4', averageGrade: 6 },
+    ]);
+  },
+
+  async down(queryInterface, Sequelize) {
+    await queryInterface.bulkDelete('city', null, {});
```
Comment (stovpak) [inline db/seeders/20220810083343-demo-cities.js]: remove seeded data, not all

File: utils/constants.js [hunk 9]
```diff
@@ -1,3 +1,7 @@
+const db = require('../db/models');
+
 module.exports = {
     port: process.env.PORT,
+    City: db.city,
```
Comment (stovpak) [inline utils/constants.js]: remove city and comment from const

File: tests/utils/dbController.js [hunk 10]
```diff
@@ -1,7 +1,9 @@
 const { Op } = require('sequelize');
-const { City, Comment } = require('../../models/models');
+const db = require('../../db/models/index');
```
Comment (stovpak) [inline tests/utils/dbController.js]: try this const { city, comment } = require('../db/models');

File: db/seeders/20220810084417-demo-commentaries.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.bulkInsert('comment', [
+      {
+ id: 1, commentText: 'Commentary for City 1', cityGrade: 10, cityId: 1,
```
Comment (stovpak) [inline db/seeders/20220810084417-demo-commentaries.js]: fix spaces

File: controllers/city.controller.js [hunk 12]
```diff
@@ -1,6 +1,9 @@
 const { isEmpty, split } = require('lodash');
 const { Op } = require('sequelize');
-const { City, Comment } = require('../models/models');
+const db = require('../db/models');
+
+const City = db.city;
+const Comment = db.comment;
 const { getPagination, getTotalPages } = require('./pagination.controller');
```
Comment (stovpak) [inline controllers/city.controller.js]: const { getPagination, getTotalPages } = require('./pagination.controller');
const City = db.city;
const Comment = db.comment;

File: db/migrations/20220808173704-create-foreign-key-for-comment.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+                onDelete: 'CASCADE',
+                allowNull: false,
+                references: { model: 'city', key: 'id' },
+            });
+        }
+    },
+
+    async down(queryInterface, Sequelize) {
+        const isCityIdExists = await queryInterface.describeTable('comment');
+        if (isCityIdExists.cityId) {
```
Comment (stovpak) [inline db/migrations/20220808173704-create-foreign-key-for-comment.js]: if isCityIdExists = null
null.cityId = error

Fix it

Comment (stovpak) [reply db/migrations/20220808173704-create-foreign-key-for-comment.js]: and u get 'comment' but variable is city

Comment (stovpak) [reply db/migrations/20220808173704-create-foreign-key-for-comment.js]: and you can have this tabel but this table can not have column 
you shoud add one more check

Comment (hhrynenko) [reply db/migrations/20220808173704-create-foreign-key-for-comment.js]: > and u get 'comment' but variable is city

Yeah, but i getting comment for city id in it

File: db/seeders/20220810084417-demo-commentaries.js (hunk 11 above)
Comment (hhrynenko) [reply db/seeders/20220810084417-demo-commentaries.js]: file is outdated. the new one have fixed spaces

**Review by stovpak — APPROVED** [review]

**Pull request merged**